### PR TITLE
Update backend guide for adding fields to Honeycomb

### PR DESCRIPTION
**Pull request by dynamike**
## Description

This PR adds documentation to `docs/backend.md`, to talk about how we can be more effective in extending Honeycomb and making it more useful.

## Code Review Verification Steps

* [x] Request review from a member of a different team.
* [x] Have the Pivotal acceptance criteria been met for this change?

### CODE REVIEW

File: docs/backend.md [hunk 1]
```diff
@@ -127,6 +129,50 @@ This is for a few reasons:
 * Any methods on struct model instances need to have pointer receivers in order to mutate the struct. This is a common point of confusion
   and an easy way to introduce bugs into a codebase.
 
+### Instrumentation
+
+[Honeycomb](https://honeycomb.io) is a hosted service used debug requests flowing through the live AWS environments. The MyMove application is configured to send events to the HoneyComb API using the [beeline-go](https://docs.honeycomb.io/getting-data-in/beelines/go-beeline/) library. The library ties into the [HTTP handler](https://golang.org/pkg/net/http/#Handler) and sends structured events to the Honeycomb API. By default, beeline-go derives fields from the incoming HTTP request. Each HTTP request maps to 1 event in Honeycomb. Below is a sample of some of the fields that are captured.
```
Comment (kilbergr) [inline docs/backend.md]: "...is a hosted service used TO debug..."

File: docs/backend.md [hunk 2]
```diff
@@ -127,6 +129,50 @@ This is for a few reasons:
...
+_Do_:
+
+* Try to come up with good field names that are descriptive to the data being represented.
+  * A good naming convention to follow is `{go pkg name}.{field name}`, so for the session package `session.office_user_id`
+* More fields are better. The Honeycomb dataset is only as powerful as the data we decide to send, so if you think it would be useful to query requests on a particular field name, add it sooner rather than later.
+* Honeycomb does a particularly good job of filtering events with unique fields like UUIDs.
+
+_Don't_:
+
+* Any fields that could potentially contain Personally Identifiable Information (PII) is a no go.
```
Comment (kilbergr) [inline docs/backend.md]: "...contain Personally Identifiable Information (PII) ARE a no go."

File: docs/backend.md [hunk 3]
```diff
@@ -127,6 +129,50 @@ This is for a few reasons:
...
+
+| field name                    | field value                          |
+|-------------------------------|--------------------------------------|
+| app.session.office_user_id    | 00000000-0000-0000-0000-000000000000 |
+| app.session.service_member_id | 2825c651-1343-40c3-a62e-2884cab25606 |
+| app.session.tsp_user_id       | 00000000-0000-0000-0000-000000000000 |
+| app.session.user_id           | 59cba4f7-99de-4d39-8b5b-3373efc7ea4c |
+
+These event is stored in the [HTTP request context](https://golang.org/pkg/net/http/#Request.Context). New fields can be added by calling [AddField](https://godoc.org/github.com/honeycombio/beeline-go#AddField).
+
```
Comment (kilbergr) [inline docs/backend.md]: Would you include an example of where this might belong in our code?

**Comment by dynamike** [comment]
@jim I'm happy to move it over.

**Pull request merged**